src/packages/shadcn-ui/ui/avatar.tsx

```diff
@@ -0,0 +1,50 @@
+"use client";
+
+import * as AvatarPrimitive from "@radix-ui/react-avatar";
+import * as React from "react";
+
+import { cn } from "@packages/utils/shadcn-utils";
+
+const Avatar = React.forwardRef<
+  React.ElementRef<typeof AvatarPrimitive.Root>,
+  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Root>
+>(({ className, ...props }, ref) => (
+  <AvatarPrimitive.Root
+    ref={ref}
+    className={cn(
+      "relative flex h-10 w-10 shrink-0 overflow-hidden rounded-full",
+      className,
+    )}
+    {...props}
+  />
+));
+Avatar.displayName = AvatarPrimitive.Root.displayName;
+
+const AvatarImage = React.forwardRef<
+  React.ElementRef<typeof AvatarPrimitive.Image>,
+  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Image>
+>(({ className, ...props }, ref) => (
+  <AvatarPrimitive.Image
+    ref={ref}
+    className={cn("aspect-square h-full w-full", className)}
+    {...props}
+  />
+));
+AvatarImage.displayName = AvatarPrimitive.Image.displayName;
+
+const AvatarFallback = React.forwardRef<
+  React.ElementRef<typeof AvatarPrimitive.Fallback>,
+  React.ComponentPropsWithoutRef<typeof AvatarPrimitive.Fallback>
+>(({ className, ...props }, ref) => (
+  <AvatarPrimitive.Fallback
+    ref={ref}
+    className={cn(
+      "flex h-full w-full items-center justify-center rounded-full bg-muted",
+      className,
+    )}
+    {...props}
+  />
+));
+AvatarFallback.displayName = AvatarPrimitive.Fallback.displayName;
```

Comment:
I personally found this code kind of ugly, code :(

Maybe we can use something like: https://www.npmjs.com/package/tailwind-styled-components

Reply:
same issue for the shared components bellow

Reply:
@marco-souza 🥶
![image](https://github.com/user-attachments/assets/1370b84d-96af-4e2d-ac8b-7cf62037df0c)

Reply:
That's what shadcn components looks like. Then maybe choose something else? or just create the components from scratch?

Reply:
Exactly, this is also code that we're not suposed to maintain, as zenha mentioned, this is shadcn's component, I don't see value on re-writing it.

Reply:
this is so sad I wanna die 😢 

- Already don't like shadcn

Reply:
Let me explain better my POV:

How can we see a `displayName` and think this is ok?

We intended to share some styles, but instead we created `forwardRef` components that feature complex and poorly readable types, utilizing spread operators - all of that do add some custom CSS 🤯 

In my opinion, this code is challenging to maintain because it requires a deep understanding of the need for `forwardRef` to customize shadcn components.

Reply:
If you guys are saying that this is the shadcn way, I won't block this PR, BUT still ugly as hell

Reply:
```suggestion
function WithStyles(ShadcnElement, styles) {
  const CustomComponent = React.forwardRef<
    React.ElementRef<typeof ShadcnElement.>,
    React.ComponentPropsWithoutRef<typeof ShadcnElement>
  >(({ className, ...props }, ref) => (
    <ShadcnElement
      ref={ref}
      className={cn(
        styles,
        className,
      )}
      {...props}
    />
  ));
  
  CustomComponent.displayName = ShadcnElement.displayName;
  
  return CustomComponent;
}
```

Then you can do something like 

```tsx
const AvatarFallback = WithStyles(
  AvatarPrimitive.Fallback,
  "flex h-full w-full items-center justify-center rounded-full bg-muted",
)
```